src/sentry/runner/initializer.py

```diff
@@ -400,16 +400,27 @@ def initialize_app(config: dict[str, Any], skip_service_validation: bool = False
     from sentry.app import env
     from sentry.runner.settings import get_sentry_conf
 
-    # Hacky workaround to dynamically set the CSRF_TRUSTED_ORIGINS for self hosted
-    if settings.SENTRY_SELF_HOSTED and not settings.CSRF_TRUSTED_ORIGINS:
-        from sentry import options
-
-        system_url_prefix = options.get("system.url-prefix")
-        if system_url_prefix:
-            settings.CSRF_TRUSTED_ORIGINS = [system_url_prefix]
-        else:
-            # For first time users that have not yet set system url prefix, let's default to localhost url
-            settings.CSRF_TRUSTED_ORIGINS = ["http://localhost:9000"]
+    if settings.SENTRY_SELF_HOSTED:
+        # XXX(isabella): Following Django upgrade, the CSRF Trusted Origins must start with a
+        # scheme. Some self-hosted users have their settings set to use old origins that are no
+        # longer valid so this updates those settings.
+        if settings.CSRF_TRUSTED_ORIGINS:
+            valid_origins = []
+            for origin in settings.CSRF_TRUSTED_ORIGINS:
+                if "://" in origin:
```

Comment:
Are we checking for `http://` or `https://` here? Would it be better to be more specific?
```
if origin.startswith(('http://', 'https://'))
```

Reply:
from my experience it has only been `http://` or `https://`, but the condition Django used to determine whether the origins were valid was just the `if "://" not in origin`. 

I could make it more specific to check that origins starting with a `.` are updated to use `https://*.<host>` in accordance with this specification: https://docs.djangoproject.com/en/4.0/releases/4.0/#csrf-trusted-origins-changes-4-0

Reply:
If Django is just checking for the `://`, then this seems fine! I also read those same release notes and thought they were checking specifically for `http://` or `https://`  

I don't think we should update the origins starting with a `.`, let the users enter in the valid origins themselves